hadoop-hdfs-project/hadoop-hdfs/src/main/java/org/apache/hadoop/hdfs/server/namenode/FSNamesystem.java

```diff
@@ -6120,7 +6121,7 @@ private void updatePipelineInternal(String clientName, ExtendedBlock oldBlock,
    */
   void registerBackupNode(NamenodeRegistration bnReg,
       NamenodeRegistration nnReg) throws IOException {
-    writeLock();
+    writeLock(FSNamesystemLockMode.FS);
```

Comment:
Why use the different locks in register and release methods?

Reply:
My bad. The FS lock is enough for `releaseBackupNode`